From d5ea4b8a0b295537d253700857c3621ba22cfe9e Mon Sep 17 00:00:00 2001
From: ChonDoit <thphantomblog@gmail.com>
Date: Mon, 31 Oct 2022 18:56:27 -0300
Subject: [PATCH] Reorder Misc Settings layout and replace checkboxes with
 toggles

Also, adapt some changes specifically for A13
| Comment FOD and A2DP items
| Enable Alternate Way to detect Headset by default

---
 app/src/main/res/xml/pref_doze.xml     |   4 +-
 app/src/main/res/xml/pref_huawei.xml   |   6 +-
 app/src/main/res/xml/pref_ims.xml      |   4 +-
 app/src/main/res/xml/pref_lenovo.xml   |   2 +-
 app/src/main/res/xml/pref_misc.xml     | 269 +++++++++++++------------
 app/src/main/res/xml/pref_nubia.xml    |  12 +-
 app/src/main/res/xml/pref_oneplus.xml  |   4 +-
 app/src/main/res/xml/pref_oppo.xml     |  10 +-
 app/src/main/res/xml/pref_qualcomm.xml |   8 +-
 app/src/main/res/xml/pref_samsung.xml  |  22 +-
 app/src/main/res/xml/pref_vsmart.xml   |   3 +-
 app/src/main/res/xml/pref_xiaomi.xml   |   7 +-
 12 files changed, 186 insertions(+), 165 deletions(-)

diff --git a/app/src/main/res/xml/pref_doze.xml b/app/src/main/res/xml/pref_doze.xml
index 6b9a84c..5835325 100644
--- a/app/src/main/res/xml/pref_doze.xml
+++ b/app/src/main/res/xml/pref_doze.xml
@@ -1,11 +1,11 @@
 <?xml version="1.0" encoding="utf-8"?>
 <PreferenceScreen xmlns:android="http://schemas.android.com/apk/res/android">
     <PreferenceCategory android:title="Doze">
-        <CheckBoxPreference
+        <SwitchPreference
             android:defaultValue="false"
             android:key="key_doze_handwave"
             android:title="Enable Handwave gesture" />
-        <CheckBoxPreference
+        <SwitchPreference
             android:defaultValue="false"
             android:key="key_doze_pocket"
             android:title="Enable out-of-pocket gesture" />
diff --git a/app/src/main/res/xml/pref_huawei.xml b/app/src/main/res/xml/pref_huawei.xml
index bedab23..ef6dcb7 100644
--- a/app/src/main/res/xml/pref_huawei.xml
+++ b/app/src/main/res/xml/pref_huawei.xml
@@ -1,17 +1,17 @@
 <?xml version="1.0" encoding="utf-8"?>
 <PreferenceScreen xmlns:android="http://schemas.android.com/apk/res/android">
     <PreferenceCategory android:title="Fingerprint">
-        <CheckBoxPreference
+        <SwitchPreference
             android:defaultValue="false"
             android:key="key_huawei_fingerprint_gestures"
             android:title="Enable Fingerprint gestures" />
     </PreferenceCategory>
     <PreferenceCategory android:title="Display">
-        <CheckBoxPreference
+        <SwitchPreference
             android:defaultValue="false"
             android:key="key_huawei_touchscreen_glove_mode"
             android:title="Enable touchscreen glove mode" />
-        <CheckBoxPreference
+        <SwitchPreference
             android:defaultValue="false"
             android:key="key_huawei_no_hwcomposer"
             android:title="Enable graphic issues workaround (recommended if ui is blurry)" />
diff --git a/app/src/main/res/xml/pref_ims.xml b/app/src/main/res/xml/pref_ims.xml
index bb98013..3c5d646 100644
--- a/app/src/main/res/xml/pref_ims.xml
+++ b/app/src/main/res/xml/pref_ims.xml
@@ -5,11 +5,11 @@
         android:key="key_ims_create_apn" />
     <Preference android:title="Download and install IMS APK"
         android:key="key_ims_install_apn" />
-    <CheckBoxPreference
+    <SwitchPreference
         android:defaultValue="false"
         android:key="key_ims_request_network"
         android:title="Request IMS network" />
-    <CheckBoxPreference
+    <SwitchPreference
         android:defaultValue="false"
         android:key="key_ims_force_enable_setting"
         android:title="Force the presence of 4G Calling setting" />
diff --git a/app/src/main/res/xml/pref_lenovo.xml b/app/src/main/res/xml/pref_lenovo.xml
index 01e1ee2..e22cfa5 100644
--- a/app/src/main/res/xml/pref_lenovo.xml
+++ b/app/src/main/res/xml/pref_lenovo.xml
@@ -2,7 +2,7 @@
 <PreferenceScreen xmlns:android="http://schemas.android.com/apk/res/android">
 
     <PreferenceCategory android:title="Other">
-        <CheckBoxPreference
+        <SwitchPreference
             android:layout_width="wrap_content"
             android:layout_height="wrap_content"
             android:defaultValue="false"
diff --git a/app/src/main/res/xml/pref_misc.xml b/app/src/main/res/xml/pref_misc.xml
index 9f67536..653af75 100644
--- a/app/src/main/res/xml/pref_misc.xml
+++ b/app/src/main/res/xml/pref_misc.xml
@@ -1,41 +1,78 @@
 <?xml version="1.0" encoding="utf-8"?>
-<PreferenceScreen
-	xmlns:android="http://schemas.android.com/apk/res/android">
-	<Preference android:title="Sponsor phh"
-        android:key="key_sponsor">
-		<intent
-		    android:action="android.intent.action.VIEW"
-		    android:data="https://github.com/sponsors/phhusson/" />
-	</Preference>
-	<PreferenceCategory android:title="Mobile network">
+<PreferenceScreen xmlns:android="http://schemas.android.com/apk/res/android">
+
+	<PreferenceCategory android:title="Show your love">
+		<Preference android:title="Sponsor phhusson"
+			android:key="key_sponsor">
+			<intent android:action="android.intent.action.VIEW"
+			android:data="https://github.com/sponsors/phhusson/" />
+		</Preference>
+	</PreferenceCategory>
+
+	<PreferenceCategory android:title="Root Solutions">
+		<Preference
+			android:title="Securize"
+			android:key="key_misc_securize"
+			android:summary="Permanently Removes root" />
+	</PreferenceCategory>
+
+	<!-- PreferenceCategory android:title="Fingerprint on Display">
+		<ListPreference
+			android:defaultValue="00ff00"
+			android:entries="@array/pref_misc_fod_color"
+			android:entryValues="@array/pref_misc_fod_color_values"
+			android:key="key_misc_fod_color"
+			android:title="Fingerprint icon color" />
+	</PreferenceCategory -->
+
+	<PreferenceCategory android:title="Mobile Network">
 		<ListPreference
-		    android:defaultValue="default"
-		    android:entries="@array/pref_misc_mobile_signal"
-		    android:entryValues="@array/pref_misc_mobile_signal_values"
-		    android:key="key_misc_mobile_signal"
-		    android:title="Signal meter method" />
+			android:defaultValue="default"
+			android:entries="@array/pref_misc_mobile_signal"
+			android:entryValues="@array/pref_misc_mobile_signal_values"
+			android:key="key_misc_mobile_signal"
+			android:title="Signal meter method" />
+		<SwitchPreference
+			android:defaultValue="false"
+			android:key="key_misc_restart_ril"
+			android:title="Automatically restart RIL" />
+		<Preference
+			android:title="@string/remove_telephony_subsystem"
+			android:key="key_misc_removetelephony" />
 	</PreferenceCategory>
+
 	<PreferenceCategory android:title="Audio">
-		<CheckBoxPreference
-            android:defaultValue="false"
-            android:key="key_huawei_headset_fix"
-            android:title="Force headset switch (Huawei default)" />
-		<CheckBoxPreference
-            android:defaultValue="false"
-            android:key="key_misc_headset_devinput"
-            android:title="Use alternate way to detect headsets" />
-		<CheckBoxPreference
+		<!-- SwitchPreference
+			android:defaultValue="false"
+			android:key="key_misc_force_a2dp_offload_disable"
+		android:title="Force-disable A2DP offload" -->
+		<SwitchPreference
+			android:defaultValue="true"
+			android:key="key_misc_headset_devinput"
+			android:title="Use alternate way to detect headsets" />
+		<SwitchPreference
 			android:defaultValue="false"
 			android:key="key_misc_disable_audio_effects"
 			android:title="Disable audio effects" />
+		<ListPreference
+			android:defaultValue="-1"
+			android:entries="@array/pref_misc_bluetooth"
+			android:entryValues="@array/pref_misc_bluetooth_values"
+			android:key="key_misc_bluetooth"
+			android:title="Bluetooth workarounds" />
+		<SwitchPreference
+			android:defaultValue="false"
+			android:key="key_huawei_headset_fix"
+			android:title="Force headset switch (Huawei default)" />
 	</PreferenceCategory>
+
 	<PreferenceCategory android:title="Display">
 		<EditTextPreference
-            android:defaultValue="-1"
-            android:key="key_misc_rounded_corners"
-	        android:title="Set rounded corners diameter"
-	        android:inputType="number"
-            android:summary="Requires a reboot" />
+			android:defaultValue="-1"
+			android:key="key_misc_rounded_corners"
+			android:title="Set rounded corners diameter"
+			android:inputType="number"
+			android:summary="Requires a reboot" />
 		<EditTextPreference
 			android:defaultValue="-1"
 			android:key="key_misc_rounded_corners_overlay"
@@ -53,113 +90,93 @@
 			android:entryValues="@array/pref_misc_display_fps_values"
 			android:key="key_misc_display_fps"
 			android:title="Force FPS" />
-        <Preference
-            android:key="key_misc_restart_systemui"
-            android:title="Restart SystemUI"
-            android:summary="Recommended after switching display resolution\nRequires root" />
-        <CheckBoxPreference
-            android:defaultValue="false"
-            android:key="key_misc_no_hwcomposer"
-            android:title="Disable HW overlays"
-            android:summary="Eats more CPU and battery" />
-        <CheckBoxPreference
-            android:defaultValue="false"
-            android:key="key_misc_aod"
-            android:title="Force allow Always-On Display" />
+		<Preference
+			android:key="key_misc_restart_systemui"
+			android:title="Restart SystemUI"
+			android:summary="Recommended after switching display resolution\nRequires root" />
+		<SwitchPreference
+			android:defaultValue="false"
+			android:key="key_misc_no_hwcomposer"
+			android:title="Disable HW overlays"
+			android:summary="Eats more CPU and battery" />
+		<SwitchPreference
+			android:defaultValue="false"
+			android:key="key_misc_aod"
+			android:title="Force allow Always-On Display" />
+		<ListPreference
+			android:defaultValue="1.86"
+			android:entries="@array/pref_misc_max_aspect_ratio_pre_o"
+			android:entryValues="@array/pref_misc_max_aspect_ratio_pre_o_values"
+			android:key="key_misc_max_aspect_ratio_pre_o"
+			android:title="Compatibility max aspect ratio" />
 	</PreferenceCategory>
-	<PreferenceCategory android:title="Backlight">
-		<CheckBoxPreference
-            android:defaultValue="false"
-            android:key="key_misc_linear_brightness"
-            android:title="Use linear screen brightness slider" />
-		<CheckBoxPreference
-            android:defaultValue="false"
-            android:key="key_misc_disable_buttons_backlight"
-            android:title="Disable backlight of hardware buttons if present"
-            android:summary="Requires a reboot" />
-		<CheckBoxPreference
-            android:defaultValue="false"
-            android:key="key_misc_backlight_scale"
-            android:title="Force alternative backlight scale" />
-        <CheckBoxPreference
-            android:defaultValue="false"
-            android:key="key_misc_minimal_brightness"
-            android:title="Allows setting brightness to the lowest possible" />
+
+	<PreferenceCategory android:title="Display Backlight">
+		<SwitchPreference
+			android:defaultValue="false"
+			android:key="key_misc_linear_brightness"
+			android:title="Use linear screen brightness slider" />
+		<SwitchPreference
+			android:defaultValue="false"
+			android:key="key_misc_backlight_scale"
+			android:title="Force alternative backlight scale" />
+		<SwitchPreference
+			android:defaultValue="false"
+			android:key="key_misc_disable_buttons_backlight"
+			android:title="Disable backlight of hardware buttons"
+			android:summary="Requires a reboot" />
+		<SwitchPreference
+			android:defaultValue="false"
+			android:key="key_misc_minimal_brightness"
+			android:title="Allows setting brightness to the lowest possible" />
 	</PreferenceCategory>
+
 	<PreferenceCategory android:title="Camera">
-		<CheckBoxPreference
+		<SwitchPreference
 			android:defaultValue="false"
 			android:key="key_misc_multi_camera"
-			android:title="Expose Aux cameras" />
-		<CheckBoxPreference
+			android:title="Expose Auxiliary cameras" />
+		<SwitchPreference
 			android:defaultValue="false"
 			android:key="key_misc_force_camera2api_hal3"
 			android:title="Force Enable Camera2API HAL3" />
-		<ListPreference
-		    android:defaultValue="-1"
-		    android:entries="@array/pref_misc_camera_timestamp"
-		    android:entryValues="@array/pref_misc_camera_timestamp_values"
-		    android:key="key_misc_camera_timestamp"
-            android:title="Force Timestamp Source"
-            android:summary="Requires a reboot" />
-  </PreferenceCategory>
-	<PreferenceCategory android:title="Other">
 		<ListPreference
 			android:defaultValue="-1"
-			android:entries="@array/pref_misc_bluetooth"
-			android:entryValues="@array/pref_misc_bluetooth_values"
-			android:key="key_misc_bluetooth"
-			android:title="Bluetooth workarounds" />
-		<ListPreference
-    	    android:defaultValue="1.86"
-    	    android:entries="@array/pref_misc_max_aspect_ratio_pre_o"
-    	    android:entryValues="@array/pref_misc_max_aspect_ratio_pre_o_values"
-    	    android:key="key_misc_max_aspect_ratio_pre_o"
-    	    android:title="Compatibility max aspect ratio" />
-		<CheckBoxPreference
-            android:defaultValue="false"
-            android:key="key_misc_force_a2dp_offload_disable"
-            android:title="Force-disable A2DP offload" />
-		<CheckBoxPreference
-            android:defaultValue="false"
-            android:key="key_misc_force_navbar_off"
-            android:title="Force navigation bar disabled"
-            android:summary="Requires a reboot" />
-		<CheckBoxPreference
-            android:defaultValue="false"
-            android:key="key_misc_storage_fuse"
-            android:title="Disable FUSE storage layer"
-            android:summary="This can (de/in)crease storage speed\nRequires a reboot" />
-		<CheckBoxPreference
-				android:defaultValue="false"
-				android:key="key_misc_restart_ril"
-				android:title="Automatically restart RIL" />
-		<CheckBoxPreference
-				android:defaultValue="false"
-				android:key="key_misc_dt2w"
-            android:title="Double tap to wake"
-            android:summary="Unlikely to work for you" />
-		<ListPreference
-		    android:defaultValue="00ff00"
-		    android:entries="@array/pref_misc_fod_color"
-		    android:entryValues="@array/pref_misc_fod_color_values"
-		    android:key="key_misc_fod_color"
-            android:title="Under-display fp color" />
-		<Preference android:title="Securize"
-            android:key="key_misc_securize" />
-		<Preference android:title="@string/remove_telephony_subsystem"
-            android:key="key_misc_removetelephony" />
-	</PreferenceCategory>
-    <PreferenceCategory android:title="Debugging">
-        <Preference android:title="Debug Sensors">
-            <intent
-                android:action="android.intent.action.VIEW"
-                android:targetPackage="me.phh.treble.app"
-                android:targetClass="me.phh.treble.app.DebugSensors" />
-        </Preference>
-		<CheckBoxPreference
-            android:defaultValue="false"
-            android:key="key_misc_remotectl"
-            android:title="Allow takeover of the device by phh for debugging" />
+			android:entries="@array/pref_misc_camera_timestamp"
+			android:entryValues="@array/pref_misc_camera_timestamp_values"
+			android:key="key_misc_camera_timestamp"
+			android:title="Force Timestamp Source"
+			android:summary="Requires a reboot" />
     </PreferenceCategory>
+
+	<PreferenceCategory android:title="Others">
+		<SwitchPreference
+			android:defaultValue="false"
+			android:key="key_misc_dt2w"
+			android:title="Double tap to wake"
+			android:summary="Unlikely to work for you" />
+		<SwitchPreference
+			android:defaultValue="false"
+			android:key="key_misc_force_navbar_off"
+			android:title="Force disable Navigation bar"
+			android:summary="Requires a reboot" />
+		<SwitchPreference
+			android:defaultValue="false"
+			android:key="key_misc_storage_fuse"
+			android:title="Disable FUSE storage layer"
+			android:summary="This can (de/in)crease storage speed\nRequires a reboot" />
+	</PreferenceCategory>
+
+	<PreferenceCategory android:title="Debugging">
+		<Preference android:title="Debug Sensors">
+			<intent android:action="android.intent.action.VIEW"
+			android:targetPackage="me.phh.treble.app"
+			android:targetClass="me.phh.treble.app.DebugSensors" />
+		</Preference>
+		<SwitchPreference
+			android:defaultValue="false"
+			android:key="key_misc_remotectl"
+			android:title="Allow takeover of the device by phh for debugging" />
+	</PreferenceCategory>
+
 </PreferenceScreen>
diff --git a/app/src/main/res/xml/pref_nubia.xml b/app/src/main/res/xml/pref_nubia.xml
index f3d330b..3d765fa 100644
--- a/app/src/main/res/xml/pref_nubia.xml
+++ b/app/src/main/res/xml/pref_nubia.xml
@@ -1,13 +1,13 @@
 <?xml version="1.0" encoding="utf-8"?>
 <PreferenceScreen xmlns:android="http://schemas.android.com/apk/res/android">
     <PreferenceCategory android:title="Touchscreen">
-        <CheckBoxPreference
+        <SwitchPreference
             android:layout_width="wrap_content"
             android:layout_height="wrap_content"
             android:defaultValue="false"
             android:key="nubia_double_tap_to_wake"
             android:title="Enable DT2W" />
-        <CheckBoxPreference
+        <SwitchPreference
             android:layout_width="wrap_content"
             android:layout_height="wrap_content"
             android:defaultValue="false"
@@ -23,7 +23,7 @@
 		    android:title="FAN speed" />
     </PreferenceCategory>
     <PreferenceCategory android:title="LEDs">
-        <CheckBoxPreference
+        <SwitchPreference
             android:layout_width="wrap_content"
             android:layout_height="wrap_content"
             android:defaultValue="false"
@@ -37,19 +37,19 @@
 		    android:title="REDMAGIC RGB" />
     </PreferenceCategory>
     <PreferenceCategory android:title="Boosts">
-        <CheckBoxPreference
+        <SwitchPreference
             android:layout_width="wrap_content"
             android:layout_height="wrap_content"
             android:defaultValue="false"
             android:key="nubia_boost_cpu"
             android:title="Boost CPU" />
-        <CheckBoxPreference
+        <SwitchPreference
             android:layout_width="wrap_content"
             android:layout_height="wrap_content"
             android:defaultValue="false"
             android:key="nubia_boost_cache"
             android:title="Boost caches" />
-        <CheckBoxPreference
+        <SwitchPreference
             android:layout_width="wrap_content"
             android:layout_height="wrap_content"
             android:defaultValue="false"
diff --git a/app/src/main/res/xml/pref_oneplus.xml b/app/src/main/res/xml/pref_oneplus.xml
index bae0fba..28224c3 100644
--- a/app/src/main/res/xml/pref_oneplus.xml
+++ b/app/src/main/res/xml/pref_oneplus.xml
@@ -16,11 +16,11 @@
     </PreferenceCategory>
 
     <PreferenceCategory android:title="Other">
-        <CheckBoxPreference
+        <SwitchPreference
             android:defaultValue="false"
             android:key="key_oneplus_usb_otg"
             android:title="Enable USB OTG" />
-        <CheckBoxPreference
+        <SwitchPreference
             android:layout_width="wrap_content"
             android:layout_height="wrap_content"
             android:defaultValue="false"
diff --git a/app/src/main/res/xml/pref_oppo.xml b/app/src/main/res/xml/pref_oppo.xml
index c4ba29d..75bbb6e 100644
--- a/app/src/main/res/xml/pref_oppo.xml
+++ b/app/src/main/res/xml/pref_oppo.xml
@@ -2,27 +2,29 @@
 <PreferenceScreen xmlns:android="http://schemas.android.com/apk/res/android">
 
     <PreferenceCategory android:title="Touchscreen">
-        <CheckBoxPreference
+        <SwitchPreference
             android:layout_width="wrap_content"
             android:layout_height="wrap_content"
             android:defaultValue="false"
             android:key="key_oppo_double_tap_to_wake"
             android:title="Enable DT2W" />
-        <CheckBoxPreference
+        <SwitchPreference
             android:layout_width="wrap_content"
             android:layout_height="wrap_content"
             android:defaultValue="false"
             android:key="key_oppo_ts_game_mode"
             android:title="Enable game mode" />
     </PreferenceCategory>
+
     <PreferenceCategory android:title="Other">
-        <CheckBoxPreference
+        <SwitchPreference
             android:defaultValue="false"
             android:key="key_oppo_usb_otg"
             android:title="Enable USB OTG" />
-        <CheckBoxPreference
+        <SwitchPreference
             android:defaultValue="false"
             android:key="key_oppo_dc_diming"
             android:title="Enable DC Diming" />
     </PreferenceCategory>
+
 </PreferenceScreen>
diff --git a/app/src/main/res/xml/pref_qualcomm.xml b/app/src/main/res/xml/pref_qualcomm.xml
index 292579d..491eca4 100644
--- a/app/src/main/res/xml/pref_qualcomm.xml
+++ b/app/src/main/res/xml/pref_qualcomm.xml
@@ -1,20 +1,20 @@
 <?xml version="1.0" encoding="utf-8"?>
 <PreferenceScreen xmlns:android="http://schemas.android.com/apk/res/android">
-    <CheckBoxPreference
+    <SwitchPreference
         android:defaultValue="false"
         android:key="key_qualcomm_alternate_mediaprofile"
         android:title="Use alternate camera profile"
         android:summary="Requires a reboot" />
-    <CheckBoxPreference
+    <SwitchPreference
         android:defaultValue="false"
         android:key="key_qualcomm_alternate_audiopolicy"
         android:title="Use alternate audio policy" />
-    <CheckBoxPreference
+    <SwitchPreference
         android:defaultValue="false"
         android:key="key_qualcomm_disable_soundvolume_effect"
         android:title="Disable soundvolume effect"
         android:summary="May fix audio distorsion" />
-    <CheckBoxPreference
+    <SwitchPreference
         android:defaultValue="false"
         android:key="key_qualcomm_disable_stereo_voip"
         android:title="Disable stereo support on VoIP"
diff --git a/app/src/main/res/xml/pref_samsung.xml b/app/src/main/res/xml/pref_samsung.xml
index c2ff352..1f66989 100644
--- a/app/src/main/res/xml/pref_samsung.xml
+++ b/app/src/main/res/xml/pref_samsung.xml
@@ -1,11 +1,11 @@
 <?xml version="1.0" encoding="utf-8"?>
 <PreferenceScreen xmlns:android="http://schemas.android.com/apk/res/android">
     <PreferenceCategory android:title="Audio">
-        <CheckBoxPreference
+        <SwitchPreference
             android:defaultValue="false"
             android:key="key_samsung_alternate_audio_policy"
             android:title="Use alternate audio policy" />
-        <CheckBoxPreference
+        <SwitchPreference
             android:defaultValue="false"
             android:key="key_samsung_audio_stereo"
             android:title="Enable stereo audio mode" />
@@ -17,29 +17,29 @@
             android:title="Set eSCO transport unit size" />
     </PreferenceCategory>
     <PreferenceCategory android:title="Screen">
-        <CheckBoxPreference
+        <SwitchPreference
             android:defaultValue="false"
             android:key="key_samsung_high_brightness"
             android:title="Enable extended brightness range" />
-        <CheckBoxPreference
+        <SwitchPreference
             android:defaultValue="false"
             android:key="key_samsung_glove_mode"
             android:title="Enable glove mode" />
     </PreferenceCategory>
     <PreferenceCategory android:title="Misc">
-        <CheckBoxPreference
+        <SwitchPreference
             android:defaultValue="false"
             android:key="key_samsung_wireless_charging_transmit"
             android:title="Enable reversed wireless charging(If applicable)" />
-        <CheckBoxPreference
+        <SwitchPreference
             android:defaultValue="false"
             android:key="key_samsung_double_tap_to_wake"
             android:title="Enable double tap to wake" />
-        <CheckBoxPreference
+        <SwitchPreference
             android:defaultValue="false"
             android:key="key_samsung_fod_single_click"
             android:title="Clicking fingerprint turns device on" />
-        <CheckBoxPreference
+        <SwitchPreference
             android:defaultValue="false"
             android:key="key_samsung_camera_ids"
             android:title="Enable access to all cameras"
@@ -52,15 +52,15 @@
 	        android:inputType="number" />
     </PreferenceCategory>
     <PreferenceCategory android:title="Workarounds">
-        <CheckBoxPreference
+        <SwitchPreference
             android:defaultValue="false"
             android:key="key_samsung_extra_sensors"
             android:title="Enable extra sensors" />
-        <CheckBoxPreference
+        <SwitchPreference
             android:defaultValue="false"
             android:key="key_samsung_colorspace"
             android:title="Enable workaround for white-ish screen" />
-        <CheckBoxPreference
+        <SwitchPreference
             android:defaultValue="false"
             android:key="key_samsung_broken_fingerprint"
             android:title="Enable workaround for broken fingerprint" />
diff --git a/app/src/main/res/xml/pref_vsmart.xml b/app/src/main/res/xml/pref_vsmart.xml
index deca41f..5ec4001 100644
--- a/app/src/main/res/xml/pref_vsmart.xml
+++ b/app/src/main/res/xml/pref_vsmart.xml
@@ -1,7 +1,8 @@
 <?xml version="1.0" encoding="utf-8"?>
 <PreferenceScreen xmlns:android="http://schemas.android.com/apk/res/android">
+
     <PreferenceCategory android:title="Display">
-        <CheckBoxPreference
+        <SwitchPreference
             android:defaultValue="false"
             android:key="key_vsmart_dt2w"
             android:title="Enable double tap to wake" />
diff --git a/app/src/main/res/xml/pref_xiaomi.xml b/app/src/main/res/xml/pref_xiaomi.xml
index 4887790..e7b0673 100644
--- a/app/src/main/res/xml/pref_xiaomi.xml
+++ b/app/src/main/res/xml/pref_xiaomi.xml
@@ -2,17 +2,18 @@
 <PreferenceScreen xmlns:android="http://schemas.android.com/apk/res/android">
 
     <PreferenceCategory android:title="Display">
-        <CheckBoxPreference
+        <SwitchPreference
             android:layout_width="wrap_content"
             android:layout_height="wrap_content"
             android:defaultValue="false"
             android:key="xiaomi_double_tap_to_wake"
-            android:title="Enable DT2W" />
-        <CheckBoxPreference
+            android:title="Enable DT2W"/>
+        <SwitchPreference
             android:layout_width="wrap_content"
             android:layout_height="wrap_content"
             android:defaultValue="false"
             android:key="xiaomi_disable_sunlight_mode"
             android:title="Disable Sunlight Mode" />
     </PreferenceCategory>
+
 </PreferenceScreen>
-- 
2.38.1

